File: Backend/src/services/interfaces/ITypesRolesServices.ts
import { typesRoles } from "../../domain/Entities/typesRoles";
import { IServices } from "./IServices";

interface ITypesRolesServices extends IServices<typesRoles>{
    GetTypeRoleByTypeName(typeName:string): Promise<typesRoles>
    VerifyTypeRoleNameExists(typeName:string): Promise<boolean>
}
export { ITypesRolesServices }

File: BackEnd/Services/Services/IServices/ITypesRolesServices.cs
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bussines.Services.IServices
{
    public interface ITypesRolesServices
    {
        Task<string> Save(TypesRoles typesroles);
        Task<string> Delete(TypesRoles typesroles);
    }
}
